shotover-proxy/src/transforms/mod.rs

```diff
@@ -265,7 +269,7 @@ impl TransformsConfig {
             TransformsConfig::DebugPrinter => Ok(Transforms::DebugPrinter(DebugPrinter::new())),
             TransformsConfig::Null => Ok(Transforms::Null(Null::default())),
             TransformsConfig::Loopback => Ok(Transforms::Loopback(Loopback::default())),
-            TransformsConfig::RedisSinkCluster(r) => r.get_source().await,
+            TransformsConfig::RedisSinkCluster(r) => r.get_source(chain_name).await,
```

Comment:
Why do we pass in the transform name on creation this one when we do it differently on runtime (which all other transforms then use)?

Transforms dynamically get their chain name here from `src/transforms/mod.rs`:


```rust
pub async fn call_next_transform(mut self) -> ChainResponse {
        if self.transforms.is_empty() {
            panic!("The transform chain does not end with a terminating transform. If you want to throw the messages away use a Null transform, otherwise use a terminating sink transform to send the messages somewhere.");
        }
        let transform = self.transforms.remove(0);

        let transform_name = transform.get_name();
        let chain_name = self.chain_name.clone();

        let start = Instant::now();
        let result = CONTEXT_CHAIN_NAME
            .scope(chain_name, transform.transform(self))
            .await;
        counter!("shotover_transform_total", 1, "transform" => transform_name);
        if result.is_err() {
            counter!("shotover_transform_failures", 1, "transform" => transform_name)
        }
        histogram!("shotover_transform_latency", start.elapsed(),  "transform" => transform_name);
        result
    }
```

It might actually be better to do do all the name / metrics setup by wrapping the chain setup function here:    

```rust
async fn _prep_transform_chain(&mut self, t: &mut TransformChain) -> Result<()> {}
```

Then actually calling that whenever we setup a transform chain (which will happen on startup).

Reply:
`async fn _prep_transform_chain(&mut self, t: &mut TransformChain) -> Result<()> {}`
 
Could then also use the same chain name injection as above?

It is fairly "magic" and non-explicit... but we should do it all the same way, rather than 2 different ways.

Reply:
Good thinking, I agree.
I'm thinking we should refactor all transform name accesses to come from CONTEXT_CHAIN_NAME in https://github.com/shotover/shotover-proxy/pull/336
Then come back to this PR once thats merged.

Reply:
oh wait, CONTEXT_CHAIN_NAME probably has the name of the chain not the transform...

Reply:
ah, so you are suggesting to create a CONTEXT_TRANSFORM_NAME?

Reply:
Oh I see whats happened. You are talking about chain names but accidentally said "transform name".

Reply:
I actually dont think we want to use CONTEXT_CHAIN_NAME here.
CONTEXT_CHAIN_NAME was a hack because we didnt want to pass the chain name really deep into the callstack.

And I think trying trying to use it here would cause problems!
Initializing chains is recursive and CONTEXT_CHAIN_NAME doesnt function like a stack so a recursive chain setup would leave the wrong value in CONTEXT_CHAIN_NAME when it finishes.

Reply:
in that case let's create a ticket to figure out "getting names into things properly and with (hopefully) one way of doing it" and I'm happy for this to get merged

Reply:
I originally attempted to use CONTEXT_CHAIN_NAME but it's only available when you're running a chain, not on initialisation.